crates/biome_module_graph/src/js_module_info/ad_hoc_scope_resolver.rs

```diff
@@ -118,6 +122,54 @@ impl TypeResolver for AdHocScopeResolver {
         GLOBAL_RESOLVER.resolve_type_of(identifier)
     }
 
+    fn resolve_import(&mut self, qualifier: &TypeImportQualifier) -> Option<Type> {
+        const MAX_DEPTH: usize = 10;
```

Comment:
Maybe let's add a comment that explains why `10`. It can be anything; however it will be helpful in case we will need to change it for a proper reason, and we will update the comment too

Reply:
I've scrapped this entire implementation for now :)